dataflowAPI/rose/semantics/BaseSemantics2.h

```diff
@@ -1712,6 +1712,11 @@ namespace rose {
                      *  number zero. The begin_bit and end_bit values must be valid for the width of @p a. */
                     virtual SValuePtr extract(const SValuePtr &a, uint64_t begin_bit, uint64_t end_bit) = 0;
 
+                    /** Extracts bits from a value.  The specified bits from begin (inclusive) through end (exclusive) are copied into
+                     *  the low-order bits of the return value (other bits in the return value are cleared). The least significant bit is
+                     *  number zero. The begin and end values must be valid for the width of @p a. */
+                    virtual SValuePtr extract(const SValuePtr &a, const SValuePtr &begin, const SValuePtr &end) = 0;
```

Comment:
This file comes from ROSE (it's now called [`RiscOperators.h`](https://github.com/rose-compiler/rose/blob/weekly/src/Rose/BinaryAnalysis/InstructionSemantics/BaseSemantics/RiscOperators.h)). Any modifications here will make it very difficult, if not impossible, to migrate toward using ROSE like a normal dependency. Is there any possibility that this functionality could be provided without modifying the ROSE classes?